optuna/trial/_trial.py

```diff
@@ -140,12 +140,20 @@ def objective(trial):
                 range.
             step:
                 A step of discretization.
+
+                .. note::
+                    The ``step`` and ``log`` arguments cannot be used at the same time. To set
+                    the ``step`` argument to a float number, set the ``log`` argument to ``False``.
             log:
                 A flag to sample the value from the log domain or not.
                 If ``log`` is true, the value is sampled from the range in the log domain.
                 Otherwise, the value is sampled from the range in the linear domain.
                 See also :func:`suggest_uniform` and :func:`suggest_loguniform`.
 
+                .. note::
+                   The ``step`` and ``log`` arguments cannot be used at the same time. To set
+                   the ``log`` argument to ``True``, set the ``step`` argument to ``None``.
```

Comment:
A minor that you can ignore but how about documenting it using `Raises:` as in https://github.com/optuna/optuna/blob/master/optuna/storages/base.py#L104? It's good practice 🙂